tools/clang/unittests/HLSL/DXIsenseTest.cpp

```diff
@@ -191,7 +198,8 @@ TEST_F(DXIntellisenseTest, InclusionWhenValidThenAvailable) {
   VERIFY_SUCCEEDED(isense->CreateIndex(&index));
   VERIFY_SUCCEEDED(isense->CreateUnsavedFile("./inc.h", unsaved_text, strlen(unsaved_text), &unsaved[0]));
   VERIFY_SUCCEEDED(isense->CreateUnsavedFile("file.hlsl", main_text, strlen(main_text), &unsaved[1]));
-  VERIFY_SUCCEEDED(index->ParseTranslationUnit("file.hlsl", nullptr, 0, &unsaved[0].p, 2,
+  IDxcUnsavedFile *unsaved_ptrs[2] = {unsaved[0].p, unsaved[1].p};
```

Comment:
would you elaborate why you needed to make this change?

Reply:
`ParseTranslationUnit` expects a pointer to a pointer to `IDxcUnsavedFile` in its fourth argument, which it goes on to treat like an array of pointers to `IDxcUnsavedFile`s. The previous code passed a pointer to a pointer to `IDxUnsavedFIle`, but it was not an array of them. It was a pointer to the location in the `CComPtr` wrapper that corresponded to the unsaved file object in the first of an array of `CComPtr`s wrapping unsaved files.

The upshot of all this is that when `ParseTranslationUnit` tries to advance to the next element in what it thinks is an array of unsaved file pointers, it ends up in part of the `CCompPtr` wrapper of either the first or the second. I have no idea how  `CComPtr`s are actually implemented in windows, but I can't really imagine that they don't have some memory allocated around their contents that would cause this same problem. That leaves me puzzled how this ever worked on Windows, but it's clear it didn't work on Linux because it was trying to treat part of the CComPtr wrapper as an instance of `IDxcUnsavedFIle`